vowpalwabbit/array_parameters_dense.h

```diff
@@ -87,6 +89,34 @@ class dense_parameters
   inline const weight& operator[](size_t i) const { return _begin[i & _weight_mask]; }
   inline weight& operator[](size_t i) { return _begin[i & _weight_mask]; }
 
+  std::unordered_map<uint64_t, std::bitset<32>> feature_bit_vector; // define the 32-bitset for each feature
+  struct tag_hash_info_dense // define struct to store the information of the tag hash
+  {
+    uint64_t tag_hash;
+    bool is_set=false;
+  };
+
+  tag_hash_info_dense tag_info_dense;
```

Comment:
same for these, move to private and rename (the struct can probably also be private)